mavisp/modules.py

```diff
@@ -2106,3 +2106,60 @@ def ingest(self, mutations):
         if len(warnings) > 0:
             raise MAVISpMultipleError(warning=warnings,
                                       critical=[])
+
+class TED(MavispModule):
+
+    module_dir = "ted"
+    name = "ted"
+
+    def ingest(self, mutations):
+        warnings = []
+
+        ted_file = os.listdir(os.path.join(self.data_dir, self.module_dir))
+        if len(ted_file) != 1:
+            this_error = f"multiple or no files found in {ted}; only one expected"
+            raise MAVISpMultipleError(warning=warnings,
+                                      critical=[MAVISpCriticalError(this_error)])
+
+        ted_file = ted_file[0]
+
+        log.info(f"parsing pfam file {ted_file}")
+
+        try:
+            ted = pd.read_csv(os.path.join(self.data_dir, self.module_dir, ted_file),
+                               sep=',', 
+                               dtype={'TED_id': str, 'TED_boundaries': str, 'CATH_label': str})
+        except Exception as e:
+            this_error = f"Exception {type(e).__name__} occurred when parsing the summary.csv file. Arguments:{e.args}"
+            raise MAVISpMultipleError(warning=warnings,
+                                        critical=[MAVISpCriticalError(this_error)])
+
+        if not set(['TED_id','TED_boundaries','CATH_label']).issubset(set(ted.columns)):
+            this_error = f"The input file doesn't have the columns expected for a TED output file"
+            raise MAVISpMultipleError(warning=warnings,
+                                      critical=[MAVISpCriticalError(this_error)])
+
+        # Split TED_boundaries into start and end residues
+        try:
+            ted[['start', 'end']] = ted['TED_boundaries'].str.split('-', expand=True).astype(int)
+        except Exception as e:
+            this_error = f"Error parsing TED_boundaries column: {e}"
+            raise MAVISpMultipleError(warning=warnings,
+                                      critical=[MAVISpCriticalError(this_error)])
+        
+        # Dictionary of muts + res_numbers
+        mutation_residues = {mut: int(mut[1:-1]) for mut in mutations}
+
+        # Map res_numbers to TED domains
+        ted_annotations = {}
+        for mutation, resn in mutation_residues.items():
+            matching_domains = ted[(ted['start'] <= resn) & (ted['end'] >= resn)]
+            ted_annotations[mutation] = ", ".join(
```

Comment:
it would be better to use another separator that is not `,`, because this already separates columns in the MAVISp output CSVs. For instane, I have used pipe (`|`) in a similar case when we annotate SLIMs